modules/backendscheduler/backendscheduler.go

```diff
@@ -290,11 +291,21 @@ func (s *BackendScheduler) UpdateJob(ctx context.Context, req *tempopb.UpdateJob
 		var (
 			metas     = s.store.BlockMetas(j.Tenant())
 			oldBlocks []*backend.BlockMeta
+			bb        uuid.UUID
+			u         backend.UUID
 		)
 
 		for _, b := range j.GetCompactionInput() {
+			u, err = backend.ParseUUID(b)
+			if err != nil {
+				level.Error(log.Logger).Log("msg", "failed to parse block ID", "block_id", b, "error", err)
+				continue
+			}
+
+			bb = (uuid.UUID)(u)
+
 			for _, m := range metas {
-				if m.BlockID.String() == b {
+				if (uuid.UUID)(m.BlockID) == bb {
 					oldBlocks = append(oldBlocks, m)
 				}
```

Comment:
If i'm not mistaken both `u` and `m.BlockID` have the type `backend.UUID`. Therefore it should be possible to simplify the code and omit the double casting to `uuid.UUID`.

Removing L294 and L305 and write the loop like this should work as well:
```go
			for _, m := range metas {
				if m.BlockID == u {
					oldBlocks = append(oldBlocks, m)
				}
			}
 ```